### PR TITLE
Irish car bomb clown car interaction

### PR DESCRIPTION
**Pull request by Fhaxaris**
## About The Pull Request
If a clown car picks up a person that drank more than 30u of Irish Car Bomb, the clown car will explode five seconds later.

## Why It's Good For The Game
feature ✔️ rich 💰 gameplay

## Changelog

:cl:
add: If a clown car picks up a person that drank more than 30u of Irish Car Bomb, the clown car will explode five seconds later in a half max cap sized explosion.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
this should be splashing the car with it, imo

**Comment by ATH1909** [comment]
this should also be, like, a proper explosion

at the very least, the guy with irish car bomb in his system should take some damage

**Comment by Fhaxaris** [comment]
I wanted it to scale with the reagent you had but I couldn't get the explosion to happen at the right loc, it would always explode where the timer and callback was created when I passed ```src``` as the explosion location,

**Comment by NanoCats** [comment]
I don't like this because the clown car is also extremely valid and you can counter it by simply laying down as is. Getting a clown car is a rare chance for us honkers :(

**Comment by squi1-rasb** [comment]
if you add a way to counter an antag item with absolutely no effort at all atleast make it a MASSIVE explosion that kills everyone inside

**Comment by Fhaxaris** [comment]
> if you add a way to counter an antag item with absolutely no effort at all atleast make it a MASSIVE explosion that kills everyone inside

based

**Comment by NanoCats** [comment]
> > if you add a way to counter an antag item with absolutely no effort at all atleast make it a MASSIVE explosion that kills everyone inside
> 
> based

maxcap or bust, gotta love when people break the clown car as if they've forgotten it explodes when you break it.

**Comment by Fikou** [comment]
i think this will be an administrative nightmare

**Comment by Fhaxaris** [comment]
> 
> 
> i think this will be an administrative nightmare

shhhhhhhhhhhhhhhhhhhhh it's logged so it's ok

File: code/modules/vehicles/cars/clowncar.dm [hunk 1]
```diff
@@ -53,16 +53,20 @@
...
 				reagent_amount += stomach.reagents.get_reagent_amount(/datum/reagent/consumable/ethanol/irishcarbomb)
-			if(reagent_amount >= 10)
-				forced_mob.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
 				stomach.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
-				message_admins("[ADMIN_LOOKUPFLW(forced_mob)] was forced into a clown car after drinking Irish Car Bomb, causing an explosion.")
-				forced_mob.log_message("was forced into a clown car after drinking Irish Car Bomb, causing an explosion.", LOG_GAME)
+			if(reagent_amount >= 10)
+				message_admins("[ADMIN_LOOKUPFLW(forced_mob)] was forced into a clown car with [reagent_amount] unit(s) of Irish Car Bomb, causing an explosion.")
+				forced_mob.log_message("was forced into a clown car with [reagent_amount] unit(s) of Irish Car Bomb, causing an explosion.", LOG_GAME)
 				audible_message(span_userdanger("You hear a rattling sound coming from the engine. That can't be good..."), null, 1)
```
Comment (ATH1909) [inline code/modules/vehicles/cars/clowncar.dm:64]: I'd recommend making this appear in runechat (so it's apparent to everyone inside) and lengthening the delay before the explosion (so that the clown and the other occupants will have time to ditch the car)

**Comment by SuperNovaa41** [comment]
Idk dude this literal terrorism we just had a bunch or contributors throw a hissy fit about this <24 hourd ago

**Comment by Naloacisamazing** [comment]
Great pr

**Comment by NanoCats** [comment]
> > i think this will be an administrative nightmare
> 
> shhhhhhhhhhhhhhhhhhhhh it's logged so it's ok

Ah yes, I see no potential issues arising with this, wait a second, I'm getting a call from **the future**

BWOINK
Admin PM From AdminMcAdminFace:

Hey so do you know why the clown car exploded, killing every able bodied crew member? The logs say you, a non antagonist, did it.

**Comment by D4C-420** [comment]
make the units in a person required something like 800 u and then its fair

**Comment by TheBonded** [comment]
> make the units in a person required something like 800 u and then its fair

Satire idea aside, increasing the max number of units to 30-50 would make it slightly more balanced, as that would likely be enough to cause the person to get drunk if they don't have the alcohol tolerance perk, which would make it harder for them to move around and could possibly make them throw up.

**Comment by Mothblocks** [comment]
Whats ur profile picture

**Comment by davethwave** [comment]
I do agree that only 10u is pretty easy to just no effort the clown car away.

**Comment by Fhaxaris** [comment]
> 
> 
> Whats ur profile picture

I like having an obscure reference as my profile picture just to see how long it takes before someone draws attention to it. Part of the fun is not telling people where it comes from. If they know, they know. If they don't, they don't. Both my last ones took two years, but this one took less than two days. Nice Pilot Red Sun profile picture, how long have you used it? Has anyone pointed it out before me?

**Comment by Toastgoats** [comment]
I think an invisible hard counter like this would be hell to balance, funny, but hard pass on this.

**Comment by itseasytosee** [comment]
> Whats ur profile picture

A google search reveals it is from a "my little pony" sex game called "banned from equestria"

**Comment by NanoCats** [comment]
> > Whats ur profile picture
> 
> I like having an obscure reference as my profile picture just to see how long it takes before someone draws attention to it. Part of the fun is not telling people where it comes from. If they know, they know. If they don't, they don't. Both my last ones took two years, but this one took less than two days. Nice PilotRedSun profile picture, how long have you used it? Has anyone pointed it out before me?

I personally was just like "oh it's a pony" and didn't think anything else of it.

File: code/modules/vehicles/cars/clowncar.dm [hunk 2]
```diff
@@ -49,6 +49,24 @@
...
+				stomach.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
+			if(reagent_amount >= 30)
+				message_admins("[ADMIN_LOOKUPFLW(forced_mob)] was forced into a clown car with [reagent_amount] unit(s) of Irish Car Bomb, causing an explosion.")
+				forced_mob.log_message("was forced into a clown car with [reagent_amount] unit(s) of Irish Car Bomb, causing an explosion.", LOG_GAME)
+				audible_message(span_userdanger("You hear a rattling sound coming from the engine. That can't be good..."), null, 1)
+				addtimer(CALLBACK(src, .proc/irish_car_bomb), 5 SECONDS)
+
+/obj/vehicle/sealed/car/clowncar/proc/irish_car_bomb()
+	src.dump_mobs(FALSE) // the mobs must be dumped or else they won't explode
+	explosion(src, 2, 5, 10, 10) // half a max cap. yep
```
Comment (Mothblocks) [inline code/modules/vehicles/cars/clowncar.dm:69]: Use named arguments here, otherwise it just looks like numbers.

File: code/modules/vehicles/cars/clowncar.dm [hunk 3]
```diff
@@ -49,6 +49,24 @@
...
+				reagent_amount += stomach.reagents.get_reagent_amount(/datum/reagent/consumable/ethanol/irishcarbomb)
+				stomach.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
+			if(reagent_amount >= 30)
+				message_admins("[ADMIN_LOOKUPFLW(forced_mob)] was forced into a clown car with [reagent_amount] unit(s) of Irish Car Bomb, causing an explosion.")
+				forced_mob.log_message("was forced into a clown car with [reagent_amount] unit(s) of Irish Car Bomb, causing an explosion.", LOG_GAME)
+				audible_message(span_userdanger("You hear a rattling sound coming from the engine. That can't be good..."), null, 1)
+				addtimer(CALLBACK(src, .proc/irish_car_bomb), 5 SECONDS)
+
+/obj/vehicle/sealed/car/clowncar/proc/irish_car_bomb()
+	src.dump_mobs(FALSE) // the mobs must be dumped or else they won't explode
```
Comment (Mothblocks) [inline code/modules/vehicles/cars/clowncar.dm:68]:
```suggestion
	dump_mobs(FALSE) // the mobs must be dumped or else they won't explode
```

Named parameter for `dump_mobs` too.

File: code/modules/vehicles/cars/clowncar.dm [hunk 4]
```diff
@@ -49,6 +49,24 @@
...
 	playsound(src, pick('sound/vehicles/clowncar_load1.ogg', 'sound/vehicles/clowncar_load2.ogg'), 75)
+	if(iscarbon(M))
+		var/mob/living/carbon/forced_mob = M
+		if(forced_mob.has_reagent(/datum/reagent/consumable/ethanol/irishcarbomb))
+			var/reagent_amount = forced_mob.reagents.get_reagent_amount(/datum/reagent/consumable/ethanol/irishcarbomb)
+			forced_mob.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
+			var/obj/item/organ/stomach/stomach = forced_mob.getorganslot(ORGAN_SLOT_STOMACH)
+			if(stomach)
+				reagent_amount += stomach.reagents.get_reagent_amount(/datum/reagent/consumable/ethanol/irishcarbomb)
+				stomach.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
```
Comment (Mothblocks) [inline code/modules/vehicles/cars/clowncar.dm:60]: Do you need all this? I think I'd rather you just `get_reagent_amount` from `forced_mob.reagents`, like everything else does, or split it into its own proc of "total" reagent amount or something.

File: code/modules/vehicles/cars/clowncar.dm [hunk 5]
```diff
@@ -49,6 +49,24 @@
 /obj/vehicle/sealed/car/clowncar/mob_forced_enter(mob/M, silent = FALSE)
 	. = ..()
 	playsound(src, pick('sound/vehicles/clowncar_load1.ogg', 'sound/vehicles/clowncar_load2.ogg'), 75)
+	if(iscarbon(M))
+		var/mob/living/carbon/forced_mob = M
+		if(forced_mob.has_reagent(/datum/reagent/consumable/ethanol/irishcarbomb))
+			var/reagent_amount = forced_mob.reagents.get_reagent_amount(/datum/reagent/consumable/ethanol/irishcarbomb)
+			forced_mob.reagents.del_reagent(/datum/reagent/consumable/ethanol/irishcarbomb)
```
Comment (Mothblocks) [inline code/modules/vehicles/cars/clowncar.dm:56]: Why bother deleting the reagent?

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I just wanted to hear you be honest about it if you lied I would've closed it

**Comment by Toastgoats** [comment]
Furthermore this is like if you made a pr that makes zombies just gib themselves if they bite you after you eat mint or some shit, or having deswords become unusable if you hit someone that has cqc.

Tldr foolproof, invisible dead-man's switches that you have no way to identify are bad, ever wonder why traitors don't get bomb implants?

File: code/modules/vehicles/cars/clowncar.dm (hunk 5 above)
Comment (Fhaxaris) [reply code/modules/vehicles/cars/clowncar.dm:56]: If you get knocked out of a clown car and picked up again, it will trigger twice otherwise.

File: code/modules/vehicles/cars/clowncar.dm (hunk 4 above)
Comment (Fhaxaris) [reply code/modules/vehicles/cars/clowncar.dm:60]: It's not necessary, and should just ```get_reagent_amount``` like everything else.

File: code/modules/vehicles/cars/clowncar.dm (hunk 5 above)
Comment (Mothblocks) [reply code/modules/vehicles/cars/clowncar.dm:56]: Sounds fair enough, make sure this is only done for their mob reagents though instead of both them and the stomach.

**Comment by Fhaxaris** [comment]
> Tldr foolproof, invisible dead-man's switches that you have no way to identify are bad, ever wonder why traitors don't get bomb implants?

It's not foolproof, there is a delay. If you notice it quickly enough, you can eject and everyone else will die, while you the clown will live. I am pretty sure everyone can agree that using the clown car is already a decidedly one-way trip. The clown will die, whether or not he drives everyone into the super-matter or a collection of welding-fuel tanks.

**Comment by Mothblocks** [comment]
Hey this isnt related to your PR but Someone deleted their comment where they said they knew where it was from and I think that's funny

**Comment by Fhaxaris** [comment]
They realized that isn't how you play the game.

File: code/modules/vehicles/cars/clowncar.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/vehicles/cars/clowncar.dm:60]: to be fair, userdanger is pretty huge

**Comment by itseasytosee** [comment]
> Hey this isnt related to your PR but Someone deleted their comment where they said they knew where it was from and I think that's funny

Didn't want to spoil the surprise.
